scipy/stats/_stats_py.py

```diff
@@ -542,21 +545,14 @@ def mode(a, axis=0, nan_policy='propagate', keepdims=None):
 
     Notes
     -----
-    The mode of object arrays is calculated using `collections.Counter`, which
-    treats NaNs with different binary representations as distinct.
-
-    .. deprecated:: 1.9.0
-        Support for non-numeric arrays has been deprecated as of SciPy 1.9.0
-        and will be removed in 1.11.0. `pandas.DataFrame.mode`_ can
-        be used instead.
```

Comment:
I will improve the error message when the user passes a non-numeric array in an immediate follow-up PR because this one is already quite large.
```python3
stats.mode(['hi', 'hi', 'bye'])
```
results in:
```
Traceback (most recent call last):

  File "C:\Users\matth\AppData\Local\Temp\ipykernel_15996\345230451.py", line 1, in <module>
    stats.mode(['hi', 'hi', 'bye'])

  File "C:\Users\matth\scipy\scipy\stats\_axis_nan_policy.py", line 513, in axis_nan_policy_wrapper
    res = hypotest_fun_out(*samples, **kwds)

  File "C:\Users\matth\scipy\scipy\stats\_stats_py.py", line 583, in mode
    return ModeResult(modes[()], counts[()])

TypeError: string indices must be integers
```